frontstage/views/passwords/forgot_password.py

```diff
@@ -43,17 +49,28 @@ def post_forgot_password():
             party_controller.reset_password_request(email)
         except ApiError as exc:
             if exc.status_code == 404:
-                logger.error('Requesting password change for email registered'
-                             ' on OAuth2 server but not in party service')
-                return render_template('passwords/forgot-password.check-email.html', email=email)
+                logger.error('Failed to retrieve details from party service')
+                return render_template('errors/404-error.html')
             raise exc
 
         logger.info('Successfully sent password change request email')
-        return redirect(url_for('passwords_bp.forgot_password_check_email'))
+        return redirect(url_for('passwords_bp.forgot_password_check_email', email=encoded_email))
 
     return render_template('passwords/forgot-password.html', form=form, email=email)
 
 
 @passwords_bp.route('/forgot-password/check-email', methods=['GET'])
 def forgot_password_check_email():
-    return render_template('passwords/forgot-password.check-email.html')
+    encoded_email = request.args.get('email', None)
+
+    if encoded_email is None:
+        logger.error('No email parameter supplied')
+        return redirect('passwords/forgot-password')
+
+    try:
+        email = url_safe_serializer.loads(encoded_email)
+    except BadSignature:
+        logger.error('Unable to decode email from URL', encoded_email=encoded_email)
+        return render_template('errors/404-error.html'), 404
```

Comment:
Use `abort(404)` here to avoid duplicating the reference to the 404-error template